src/main/java/com/mateacademy/patterny/proxy/ProxyProject.java

```diff
@@ -0,0 +1,20 @@
+package com.mateacademy.patterny.proxy;
+
+public class ProxyProject implements Project {
+
+    private String url;
+    private RealProject realProject;
+
+    public ProxyProject(String url) {
+        this.url = url;
+    }
+
+
+    @Override
+    public void run() {
+        if (realProject == null) {
```

Comment:
And what beatifull add your proxy?

Reply:
I showed the goal proxy: replacing another object and controlling access to it

Reply:
we have some Prodject, we have implementation RealProject and we have implementation ProxyProject and we hahe a client project: ProjectRunner which  causes ProxyProject, which contains as an element RealProject  and inside yourself creates this real project